src/blueprints/tv2/action-factories/tv2-video-clip-action-factory.ts

```diff
@@ -0,0 +1,175 @@
+  }
+
+  public getMutateActionMethods(action: Action): MutateActionMethods | undefined {
+    if (action.id.includes(VIDEO_CLIP_AS_NEXT_ACTION_ID_PREFIX)) {
+      return {
+        type: MutateActionType.MEDIA,
+        getMediaId: () => action.name,
+        updateActionWithMedia: (action, media) => this.updateVideoClipAction(action, media)
+      }
+    }
+  }
+
+  private updateVideoClipAction(action: Action, media?: Media): Action {
+    const videoClipAction: Tv2VideoClipAction = action as Tv2VideoClipAction
+    if (videoClipAction.metadata.contentType !== Tv2ActionContentType.VIDEO_CLIP) {
+      console.error('Can\'t update VideoClipAction. Action is not a VideoClipAction')
+      return action
+    }
+
+    videoClipAction.data.partInterface.expectedDuration = media?.duration
+      ? Math.max((media.duration * 1000) - videoClipAction.metadata.configuredVideoClipPostrollDuration, 0)
+      : videoClipAction.data.partInterface.expectedDuration
+
+    const mediaPlayerSession: string = `${action.id}_${Date.now()}`
+    videoClipAction.data.pieceInterfaces.map(pieceInterface => pieceInterface.timelineObjects.map(timelineObject => {
+      const blueprintTimelineObject: Tv2BlueprintTimelineObject = timelineObject as Tv2BlueprintTimelineObject
+      blueprintTimelineObject.metaData = {
+        ...blueprintTimelineObject.metaData,
+        mediaPlayerSession
+      }
+    }))
+
+    return videoClipAction
+  }
+
+  public createVideoClipActions(configuration: Tv2BlueprintConfiguration, videoClipData: Tv2VideoClipData[]): Tv2VideoClipAction[] {
+    return videoClipData.map(videoClip => this.createInsertVideoClipAsNextAction(configuration, videoClip))
+  }
+
+  private createInsertVideoClipAsNextAction(configuration: Tv2BlueprintConfiguration, videoClipData: Tv2VideoClipData): Tv2VideoClipAction {
+    const partId: string = 'videoClipInsertAction'
+    const partInterface: PartInterface = this.createPartInterface(partId, videoClipData)
+    return {
+      id: `${VIDEO_CLIP_AS_NEXT_ACTION_ID_PREFIX}_${videoClipData.fileName}`,
+      name: videoClipData.name,
+      type: PartActionType.INSERT_PART_AS_NEXT,
+      data: {
+        partInterface,
+        pieceInterfaces: [
+          this.createProgramPieceInterface(configuration, partId, videoClipData)
+        ]
+      },
+      metadata: {
+        contentType: Tv2ActionContentType.VIDEO_CLIP,
+        fileName: videoClipData.fileName,
+        configuredVideoClipPostrollDuration: configuration.studio.ServerPostrollDuration
+      }
+    }
+  }
+
+  private createProgramPieceInterface(configuration: Tv2BlueprintConfiguration, partId: string, videoClipData: Tv2VideoClipData): PieceInterface {
+    const metadata: Tv2PieceMetadata = {
+      sisyfosPersistMetaData: {
+        sisyfosLayers: [],
+        acceptsPersistedAudio: videoClipData.adLibPix &&  videoClipData.isVoiceOver
+      }
+    }
+
+    const videoSwitcherEnable: TimelineEnable = {
```

Comment:
Didn't we aggre to use `Mixer` instead of `Switcher`, thus this should be named `videoMixerEnable`?

Reply:
We did. This is a slip up :) 
Good catch.